src/stim/mem/simd_bits.test.cc

```diff
@@ -253,6 +317,23 @@ TEST_EACH_WORD_SIZE_W(simd_bits, left_shift_assignment, {
     }
     m0 <<= 488;
     ASSERT_TRUE(!m0.not_zero());
+    simd_bits<W> m1(448), m2(448);
+    m1.randomize(448, SHARED_TEST_RNG());
+    m2 = m1;
+    m1 <<= 1;
+    for (size_t k = 0; k < 448 - 1; k++) {
+        ASSERT_EQ(m1[k + 1], m2[k]);
+    }
+    m1.randomize(448, SHARED_TEST_RNG());
+    m2 = m1;
+    m1 <<= 10;
```

Comment:
Also try a random offset from [0, num_bits_padded()] and verify bits up to the padding not just up to the requested length.